expat/lib/xmlparse.c

```diff
@@ -4979,18 +4976,22 @@ doProlog(XML_Parser parser,
     case XML_ROLE_GROUP_OPEN:
       if (parser->m_prologState.level >= parser->m_groupSize) {
         if (parser->m_groupSize) {
-          char *temp = (char *)REALLOC(parser, parser->m_groupConnector, parser->m_groupSize *= 2);
-          if (temp == NULL) {
-            parser->m_groupSize /= 2;
-            return XML_ERROR_NO_MEMORY;
+          {
+            char * const new_connector = (char *)REALLOC(parser,
```

Comment:
As mentioned in pull request #269, we can use temp variable itself for reuse by defining it void pointer to avoid extra allocation

Reply:
My reply at https://github.com/libexpat/libexpat/pull/269/files#r300837217 just now addresses this concern.